force-app/main/default/classes/SOQL.cls

```diff
@@ -314,28 +423,36 @@ public with sharing class SOQL {
      */
     public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
         // Query the original Opportunity Line Items
-
+        List<OpportunityLineItem> originalLineItems = [SELECT Id, Name, Quantity, UnitPrice, OpportunityId, PriceBookEntryId FROM OpportunityLineItem WHERE OpportunityId = :origOppId];
         // Create a list to hold the new line items
-
+        List<OpportunityLineItem> lineItemsToInsert = new List<OpportunityLineItem>();
         // Generate a random number between 1 and 3
-
+        Integer randomNumber = Integer.valueOf(Math.floor(Math.random() * 3) + 1);
         // Loop through each cloned Opportunity Id
-
+        for(Id clonedOpp : cloneOppIds) {
             // Loop through each original line item
-
-                // Clone the original line item and add the random number to the quantity
-
-        // Insert the new line items
-    }
+          for(OpportunityLineItem originalLineItem : originalLineItems) {
+            OpportunityLineItem newOli = originalLineItem.clone(false, false, false, false);         // Clone the original line item and add the random number to the quantity
```

Comment:
Nice job using clone. Is there another way of doing this by removing the id and just copying over the field? Sometimes you'll need to do manual clones in your code.